django/db/backends/sqlite3/schema.py

```diff
@@ -216,8 +216,9 @@ def is_self_referential(f):
         if getattr(create_field, "primary_key", False) or (
             alter_field and getattr(alter_field[1], "primary_key", False)
         ):
+            _, new_field = alter_field
             for name, field in list(body.items()):
-                if field.primary_key:
+                if field.primary_key and name != new_field.name:
```

Comment:
Do not remove the old primary key when an altered field that introduces a primary key is the same field.

Reply:
Maybe that's worth a code comment.

Reply:
I added a comment.